httpclient5/src/main/java/org/apache/hc/client5/http/impl/nio/PoolingAsyncClientConnectionManager.java

```diff
@@ -183,6 +184,12 @@ public void execute(final Boolean result) {
 
                                 })));
                             } else {
+                                if (!connection.isOpen()) {
```

Comment:
This seems to be half the implementation of the blocking connection `isStale()` check